torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -98,6 +98,20 @@ std::vector<at::Tensor> ProcessGroupMPI::WorkMPI::result() {
   return outputTensors_;
 }
 
+c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupMPI::WorkMPI::getFuture() {
+  return future_;
+}
+
+void ProcessGroupMPI::WorkMPI::finishCompleteErrorFuture(std::exception_ptr eptr) {
+  future_->setError(eptr);
+  finish(eptr);
+}
+
+void ProcessGroupMPI::WorkMPI::finishCompleteFuture() {
```

Comment:
nit: curious about the naming. Why do we need both finish and complete in the function name? Do you mean `finishCompletedFuture`, or will `markCompleted` work?

Reply:
I meant that it does both, let me just call it finishWorkMPI() so that it's finishes whatever belongs to WorkMPI.